lux/vislib/altair/Histogram.py

```diff
@@ -45,48 +44,52 @@ def initialize_chart(self):
 
         x_min = self.vis.min_max[msr_attr.attribute][0]
         x_max = self.vis.min_max[msr_attr.attribute][1]
+        x_range = abs(x_max - x_min)
 
         if isinstance(msr_attr.attribute, str):
             msr_attr.attribute = msr_attr.attribute.replace(".", "")
 
         colval = self.vis.data[msr_attr.attribute]
-        x_range = abs(max(colval) - min(colval))
-        plot_range = abs(x_max - x_min)
-        markbar = x_range / plot_range * 12
+        markbar = get_bin_size(self.data[msr_attr.attribute])
+        # Default when IQR was too small
+        if markbar < (x_range / 24):
+            markbar = (x_max - x_min) / 12
 
         self.data = AltairChart.sanitize_dataframe(self.data)
+        end_attr_abv = str(msr_attr.attribute) + "_end"
+        self.data[end_attr_abv] = self.data[str(msr_attr.attribute)].apply(lambda x: x + markbar)
 
         axis_title = f"{msr_attr_abv} (binned)"
         if msr_attr.attribute == " ":
             axis_title = "Series (binned)"
         if measure.channel == "x":
             chart = (
                 alt.Chart(self.data)
-                .mark_bar(size=markbar)
+                .mark_bar()
                 .encode(
-                    alt.X(
+                    x=alt.X(
                         str(msr_attr.attribute),
                         title=axis_title,
                         bin=alt.Bin(binned=True),
                         type=msr_attr.data_type,
-                        axis=alt.Axis(labelOverlap=True, title=axis_title),
-                        scale=alt.Scale(domain=(x_min, x_max)),
+                        axis=alt.Axis(title=axis_title),
+                        scale=alt.Scale(domain=[x_min, x_max + markbar]),
```

Comment:
Why do you need to add the mark bar to the max? Shouldn't the bins be between min and max?

Reply:
I ran into an issue where the last bin would exceed the maximum range. For example, if you take a look at the three graphs below after running, 
```
df = pd.read_csv("https://github.com/covidvis/covid19-vis/blob/master/data/interventionFootprintByState.csv?raw=True",index_col=0)
df
```
I get the following output. 
![Screen Shot 2021-03-02 at 10 37 09 AM](https://user-images.githubusercontent.com/44735047/109697628-4d9b9800-7b43-11eb-9ace-f4af0309dd58.png)

Thus, I added markbar to account for the bin width. ?

Reply:
Something looks wrong in those charts. The ticks are not between the bars as they should be. 

Can you send me one of those specs as a Vega-Lite spec? I can give you better feedback that way.

Reply:
I think I've figured it for Distribution charts following from my realization [here](https://github.com/lux-org/lux/pull/285#discussion_r585924162). Specifically, ticks are now between the bars. Does this look better? 
![Screen Shot 2021-03-02 at 1 59 25 PM](https://user-images.githubusercontent.com/44735047/109721058-b8f36300-7b5f-11eb-983d-298c1dd4d56c.png)

Not super sure what a "Vega-Lite spec" is, but here is the output from running `to_Altair()` on one of the visualizations: 
```
chart = alt.Chart(visData).mark_bar(size=3.8899999999999997).encode(
    alt.X('severityScore', title='severityScore (binned)',bin=alt.Bin(binned=True), type='quantitative', axis=alt.Axis(labelOverlap=True, title='severityScore (binned)'), scale=alt.Scale(domain=(0.1, 39.0))),
    alt.Y("Number of Records", type="quantitative")
)
```